extension/jaegerremotesampling/internal/remote_strategy_cache.go

```diff
@@ -0,0 +1,123 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package internal // import "github.com/open-telemetry/opentelemetry-collector-contrib/extension/jaegerremotesampling/internal"
+
+import (
+	"sync"
+	"time"
+
+	"github.com/jaegertracing/jaeger/thrift-gen/sampling"
+)
+
+type serviceStrategyCache interface {
+	get(serviceName string) (*sampling.SamplingStrategyResponse, bool)
+	put(serviceName string, response *sampling.SamplingStrategyResponse)
+	Close() error
+}
+
+// serviceStrategyCacheEntry is a timestamped sampling strategy response
+type serviceStrategyCacheEntry struct {
+	retrievedAt      time.Time
+	strategyResponse *sampling.SamplingStrategyResponse
+}
+
+// serviceStrategyTTLCache is a naive in-memory TTL serviceStrategyTTLCache of service-specific sampling strategies returned from the remote source.
+// Each cached item has its own TTL used to determine whether it is valid for read usage (based on the time of write).
+type serviceStrategyTTLCache struct {
+	itemTTL time.Duration
+	items   map[string]serviceStrategyCacheEntry
+	clockFn func() time.Time
+
+	stopCh chan struct{}
+	mu     sync.RWMutex
```

Comment:
More of a nit, what is the mutex protecting? Also, I would suggest naming it `rw` instead of `mu` since that is the default for `RWMutex`.

More of a stylistic thing is ordering the fields so that mutex sits as a hat on top of fields it protects, like:

```golang
type serviceStrategyTTLCache struct {
  itemTTL time.Duration
  clockFn func()time.Time

  rw sync.RWMutext
  stopChn chan struct{}
  items map[string]...
}
```